packages/atomic/src/components/common/item-link/item-link.spec.ts

```diff
@@ -0,0 +1,432 @@
+
+      const link = element.querySelector('a');
+      expect(link).toBeInTheDocument();
+      expect(link).toHaveTextContent('');
+    });
+  });
+
+  describe('prop handling', () => {
+    it('should apply className when provided', async () => {
+      const element = await renderComponent({
+        className: 'custom-class another-class',
+      });
+
+      const link = element.querySelector('a');
+      expect(link).toHaveClass('custom-class', 'another-class');
+    });
+
+    it('should apply part attribute when provided', async () => {
+      const element = await renderComponent({
+        part: 'custom-part',
+      });
+
+      const link = element.querySelector('a');
+      expect(link).toHaveAttribute('part', 'custom-part');
+    });
+
+    it('should apply title when provided', async () => {
+      const element = await renderComponent({
+        title: 'Custom Title',
+      });
+
+      const link = element.querySelector('a');
+      expect(link).toHaveAttribute('title', 'Custom Title');
+    });
+
+    it('should apply rel attribute when provided', async () => {
+      const element = await renderComponent({
+        rel: 'noopener noreferrer',
+      });
+
+      const link = element.querySelector('a');
+      expect(link).toHaveAttribute('rel', 'noopener noreferrer');
+    });
+
+    it('should apply custom target when provided', async () => {
+      const element = await renderComponent({
+        target: '_blank',
+      });
+
+      const link = element.querySelector('a');
+      expect(link).toHaveAttribute('target', '_blank');
+    });
+
+    it('should default to _self target when not provided', async () => {
+      const element = await renderComponent({});
+
+      const link = element.querySelector('a');
+      expect(link).toHaveAttribute('target', '_self');
+    });
+
+    it('should apply tabIndex when provided', async () => {
+      const element = await renderComponent({
+        tabIndex: 5,
+      });
+
+      const link = element.querySelector('a');
+      expect(link).toHaveAttribute('tabindex', '5');
+    });
+
+    it('should not apply tabindex when not provided', async () => {
+      const element = await renderComponent({});
+
+      const link = element.querySelector('a');
+      expect(link).not.toHaveAttribute('tabindex');
+    });
+
+    it('should apply aria-hidden when ariaHidden is true', async () => {
+      const element = await renderComponent({
+        ariaHidden: true,
+      });
+
+      const link = element.querySelector('a');
+      expect(link).toHaveAttribute('aria-hidden', 'true');
+    });
+
+    it('should not apply aria-hidden when ariaHidden is false', async () => {
+      const element = await renderComponent({
+        ariaHidden: false,
+      });
+
+      const link = element.querySelector('a');
+      expect(link).not.toHaveAttribute('aria-hidden');
+    });
+  });
+
+  describe('custom attributes', () => {
+    it('should apply custom attributes when provided', async () => {
```

Comment:
What happens when you pass a custom attribute that conflics with an existing prop/attribute?

E.g., you pass an `href` prop, but you also pass an `href` custom attribute? Which one takes precedence?